<launch>

    <!--change accordingly to the USB camera  -->
    <!-- <arg name="video_device" default="/dev/video1" />

    <node
        name="usb_cam" pkg="usb_cam" type="usb_cam_node" output="screen">
         <param name="video_device" value="$(arg video_device)" />
         <param name="image_width" value="640" />
         <param name="image_height" value="480" />
    </node>

    <node
        name="vision" pkg="pinteraction" type="vision.cpp" respawn="true">
    </node> -->

    <group>
    <arg name="color" default="red"/>
    <arg name="debug" value="true"/>
      <node name="vision_target" pkg="pinteraction" type="vision_target" required="true" output="screen" >
        <param name="subscribed_image_topic" value="image_rect_color"/>
        <param name="object_color" value="$(arg color)"/>
        <param name="debug" value="$(arg debug)"/>
        <param name="published_topic" value="center"/>
      </node>
    </group>

    <arg name="video_device" default="/dev/video0" />
    <arg name="camera_frame_id" default="camera_link" />
    <!--arg name="namespace" default="front_cam" /-->
    <arg name="camera_info" default="camera_info" />
    <arg name="camera_image_raw" default="image_raw" />
    <arg name="camera_model" default="logitech_c150" />
    <arg name="debug" value="false" />

    <group>
        <node name="usb_cam" pkg="usb_cam" type="usb_cam_node" clear_params="true" output="screen" >

            <param name="camera_info_url" type="string" value="package://pinteraction/cfg/$(arg camera_model).yaml" />
            <remap from="usb_cam/image_raw" to="$(arg camera_image_raw)" />
            <remap from="usb_cam/camera_info" to="$(arg camera_info)" />

            <param name="video_device" value="$(arg video_device)" />
            <param name="image_width" value="640" />
            <param name="image_height" value="480" />
            <param name="framerate" value="30" />
            <!--param name="contrast" value="128"/>
            <param name="brightness" value="128" />
            <param name="saturation" value="128" /-->
            <param name="pixel_format" value="yuyv" />
            <param name="camera_frame_id" value="$(arg camera_frame_id)" />
            <param name="io_method" value="mmap" />
        </node>

        <node name="image_proc" pkg="image_proc" type="image_proc" clear_params="true" >
        </node>

        <node name="image_view" pkg="image_view" type="image_view" respawn="false" output="screen" if="$(arg debug)" >
            <remap from="image" to="image_rect_color" />
            <param name="autosize" value="true" />
        </node>
    </group>

    <node
        name="vision_actuate" pkg="pinteraction" type="vision_actuate.py" respawn="true" output="screen">
    </node>

    <!--ARDUINO  -->
    <node
        name="serial_node_1" pkg="rosserial_python" type="serial_node.py" args="/dev/Arduino1" respawn="true">
    </node>

    <node
        name="serial_node_2" pkg="rosserial_python" type="serial_node.py" args="/dev/Arduino2" respawn="true">
    </node>

    <node
        name="serial_node_3" pkg="rosserial_python" type="serial_node.py" args="/dev/Arduino3" respawn="true">
    </node>

    <node
        name="serial_node_4" pkg="rosserial_python" type="serial_node.py" args="/dev/Arduino4" respawn="true">
    </node>

    <node
        name="serial_node_5" pkg="rosserial_python" type="serial_node.py" args="/dev/Arduino5" respawn="true">
    </node>

    <node
        name="serial_node_6" pkg="rosserial_python" type="serial_node.py" args="/dev/Arduino6" respawn="true">
    </node>

    <node
        name="serial_node_7" pkg="rosserial_python" type="serial_node.py" args="/dev/Arduino7" respawn="true">
    </node>

    <node
        name="serial_node_8" pkg="rosserial_python" type="serial_node.py" args="/dev/Arduino8" respawn="true">
    </node>

    <node
        name="serial_node_9" pkg="rosserial_python" type="serial_node.py" args="/dev/Arduino9" respawn="true">
    </node>

    <node
        name="serial_node_10" pkg="rosserial_python" type="serial_node.py" args="/dev/Arduino10" respawn="true">
    </node>

    <node
        name="serial_node_r" pkg="rosserial_python" type="serial_node.py" args="/dev/ArduinoR" respawn="true">
    </node>

</launch>